src/exengine/kernel/ex_event_base.py

```diff
@@ -46,11 +46,24 @@ class ExecutorEvent(ABC, metaclass=_ExecutorEventMeta):
     def __init__(self, *args, **kwargs):
         super().__init__()
         self._num_retries_on_exception = 0
+        self._priority = 1 # lower number means higher priority
         self._finished = False
         self._initialized = False
         # Check for method-level preferred thread name first, then class-level
         self._thread_name = getattr(self.execute, '_thread_name', None) or getattr(self.__class__, '_thread_name', None)
 
+    def __lt__(self, other) -> bool:
```

Comment:
Are you thinking to make the events in a priority queue? So you could submit an event that gets jumped ahead of existing ones? I had a mechanism for this that perhaps you saw in `executor.submit`, but I think this makes more sense